pilot/pkg/model/sidecar.go

```diff
@@ -144,17 +156,34 @@ func DefaultSidecarScopeForNamespace(ps *PushContext, configNamespace string) *S
 	defaultEgressListener.virtualServices = ps.VirtualServices(&dummyNode, meshGateway)
 
 	out := &SidecarScope{
-		EgressListeners:       []*IstioEgressListenerWrapper{defaultEgressListener},
-		services:              defaultEgressListener.services,
-		destinationRules:      make(map[host.Name]*Config),
-		namespaceDependencies: make(map[string]struct{}),
+		EgressListeners:             []*IstioEgressListenerWrapper{defaultEgressListener},
+		services:                    defaultEgressListener.services,
+		destinationRules:            make(map[host.Name]*Config),
+		serviceDependencies:         make(map[host.Name]struct{}),
+		virtualServiceDependencies:  make(map[string]struct{}),
+		destinationRuleDependencies: make(map[string]struct{}),
+		namespaceDependencies:       make(map[string]struct{}),
 	}
+
 	// Now that we have all the services that sidecars using this scope (in
 	// this config namespace) will see, identify all the destinationRules
 	// that these services need
 	for _, s := range out.services {
 		out.destinationRules[s.Hostname] = ps.DestinationRule(&dummyNode, s)
 		out.namespaceDependencies[s.Attributes.Namespace] = struct{}{}
+		out.serviceDependencies[s.Hostname] = struct{}{}
+	}
+
+	for _, dr := range out.destinationRules {
+		if dr != nil {
+			out.destinationRuleDependencies[dr.Name] = struct{}{}
```

Comment:
name is not sufficient. if sidecar imports dest rules of same name from multiple namespaces (like "default"), then this will have collisions

Reply:
Yeah. Please see the comment from howardjohn above. We've discussed whether to use a data structure to describe a "changed config" or just combine the data into a string.